selfdrive/locationd/SConscript

```diff
@@ -31,7 +31,7 @@ if arch != "Darwin":
   lenv["LINKFLAGS"] += ["-Wl,--no-as-needed"]
 
 lenv["LIBPATH"].append(Dir(rednose_gen_dir).abspath)
-lenv["RPATH"].append(Dir(rednose_gen_dir).abspath)
+lenv["RPATH"].append(lenv.Literal(f'\\$$ORIGIN/{rednose_gen_dir}'))
```

Comment:
oh i guess this makes it robust to launching from different places?

Reply:
Yes, that but openpilot doesn't compile if used relpath since RPATH is used during runtime which means the path has to be relative to any program from anywhere, but if we use relpath then the path will be relative to where Sconstruct file is (/tmp/openpilot in docker). If a file calls (locationd in this case) then RPATH is not relative to that file but to the root workspace (/tmp/openpilot). So $ORIGIN env variable makes it possible to make the search path for RPATH relative to where the program was compiled.